Projects/Server/Network/Packets/IncomingAccountPackets.cs

```diff
@@ -185,6 +188,150 @@ public static void CreateCharacter(NetState state, CircularBufferReader reader,
+        if (state.NewCharacterCreation)
+        {
+            skills[3] = new SkillNameValue((SkillName)reader.ReadByte(), reader.ReadByte());
+        }
+
+        var unknown2 = reader.ReadAscii(25); // Pack of 0x00
+        int unk7 = reader.ReadByte(); // Another 0x00
+
+        int hairHue = reader.ReadInt16();
+        int hairID = reader.ReadInt16();
+
+        int unk8 = reader.ReadByte();
+        int unk9 = reader.ReadInt32();
+        int unk10 = reader.ReadByte();
+        int shirtHue = reader.ReadInt16();
+        int shirtID = reader.ReadInt16();
+        int unk13 = reader.ReadByte();
+        int faceColor = reader.ReadInt16();
+        int faceID = reader.ReadInt16();
+        int unk14 = reader.ReadByte();
+        int beardHue = reader.ReadInt16();
+        int beardID = reader.ReadInt16();
+
+        int cityIndex = 0; // Obsolete
+        int pantsHue = shirtHue; // Obsolete
+        Race race = null;
+        bool female = false;
+
+        female = (gender != 0);
+        race = Race.Races[(byte)(((genderRace - 1)))]; //SA client sends race packet one higher than KR, so this is neccesary
+        if (race == null)
+            race = Race.DefaultRace;
+
+
+        CityInfo[] info = state.CityInfo;
+        var a = state.Account;
+
+        if (clientFlags > 0)
+            flags = clientFlags;
+
+        if (info == null || a == null || cityIndex < 0 || cityIndex >= info.Length)
+        {
+            state.Disconnect("Invalid city selected during character creation.");
+            return;
+        }
+        else
+        {
+            // Check if anyone is using this account
+            for (int i = 0; i < a.Length; ++i)
+            {
+                Mobile check = a[i];
+
+                if (check != null && check.Map != Map.Internal)
+                {
+                    state.LogInfo("Account in use");
+                    state.SendPopupMessage(PMMessage.CharInWorld);
+                    return;
+                }
+            }
+
+            state.Flags = (ClientFlags)flags;
+
+            var args = new CharacterCreatedEventArgs(
+                state,
+                a,
+                name,
+                female,
+                hue,
+                stats,
+                info[cityIndex],
+                skills,
+                shirtHue,
+                pantsHue,
+                hairID,
+                hairHue,
+                beardID,
+                beardHue,
+                profession,
+                race
+            );
+            state.SendClientVersionRequest();
+
+            state.BlockAllPackets = true;
+
+            EventSink.InvokeCharacterCreated(args);
+
+            Mobile m = args.Mobile;
+
+            if (m != null)
+            {
+                state.Mobile = m;
+                m.NetState = state;
+
+                state.BlockAllPackets = false;
+                DoLogin(state, m);
```

Comment:
We need an if check to see if we have the client version. Otherwise we will need to use the LoginTimer.

Reply:
Solved